### PR TITLE
implement basic api: insert events, list events, get event by id

### PR DESCRIPTION
**Pull request by ThibaultLaurens**
coming next:
- add unit test for api.go
- custom errors from store to api (not found etc.)
- change insert events in mongo.go to upsert events
- labels validation rules on insert events (tenant required etc.)

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -1 +1,133 @@
...
+	w.Write(response)
+}
+
+func (api *API) listEventsHandler(w http.ResponseWriter, r *http.Request) {
+	labels := model.LabelSet{}
+
+	queries := r.URL.Query()
+	query, ok := queries["labels"]
+	if ok {
+		if err := labels.DecodeString(query[0]); err != nil {
```
Comment (alexdias) [inline api/api.go]: `if ok && err := labels.DecodeString(query[0]); err != nil` ?

File: api/api.go [hunk 2]
```diff
@@ -1 +1,133 @@
...
+			api.respondWithError(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+	}
+
+	events, err := api.store.ListEvents(labels)
+	if err != nil {
```
Comment (alexdias) [inline api/api.go]: Let's log the error here?

File: api/api.go [hunk 3]
```diff
@@ -1 +1,133 @@
...
+	}
+	api.respondWithJSON(w, http.StatusOK, events)
+}
+
+func (api *API) getEventHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	id := vars["id"]
+
+	event, err := api.store.GetEvent(model.Fingerprint(id))
+	if err != nil {
```
Comment (alexdias) [inline api/api.go]: Same as above

File: api/api.go (hunk 2 above)
Comment (alexdias) [reply api/api.go]: Nevermind, missed the respondWithError method

File: api/api.go [hunk 4]
```diff
@@ -1 +1,133 @@
...
+		validEvents = append(validEvents, event)
+	}
+
+	if _, err := api.store.InsertEvents(validEvents); err != nil {
+		api.respondWithError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	if len(validationErrors) > 0 {
+		api.respondWithError(w, http.StatusInternalServerError, validationErrors[0].Error())
```
Comment (alexdias) [inline api/api.go]: Should this be http.StatusBadRequest instead?

File: api/api.go [hunk 5]
```diff
@@ -1 +1,133 @@
...
+
+	router.Use(api.middleware.MonitorRequests)
+	router.Use(api.middleware.LogRequests)
+
+	return router
+}
+
+func (api *API) respondWithError(w http.ResponseWriter, code int, msg string) {
+	api.logger.Errorw("api error", "error", msg)
+	http.Error(w, msg, http.StatusInternalServerError)
```
Comment (alexdias) [inline api/api.go]: Code doesn't seem to be in use here

File: api/api.go (hunk 1 above)
Comment (ThibaultLaurens) [reply api/api.go]: wont compile :/

File: api/api.go (hunk 4 above)
Comment (ThibaultLaurens) [reply api/api.go]: it should, will add it in next commit. next pr will improve error handling in there

File: api/api.go (hunk 5 above)
Comment (ThibaultLaurens) [reply api/api.go]: nice spot cheers

File: cmd/messenger/main.go [hunk 6]
```diff
@@ -18,8 +24,11 @@ var Config = viper.New()
 func init() {
 	BindEnvAndSetDefault("debug", false, "debug flag")
 	BindEnvAndSetDefault("data_store", "memory", "data store type: [memory|mongo]")
+	BindEnvAndSetDefault("event_end_time", 5, "default end time for events")
```
Comment (alexdias) [inline cmd/messenger/main.go]: I think we should clarify that this is the number of minutes from the time it is processed

Comment (ThibaultLaurens) [reply cmd/messenger/main.go]: oh yeah

File: api/api.go (hunk 1 above)
Comment (alexdias) [reply api/api.go]: Apologies:
`if err := labels.DecodeString(query[0]);  ok &&  err != nil`

Comment (alexdias) [reply api/api.go]: But nevermind

File: cmd/messenger/main.go [hunk 7]
```diff
@@ -75,3 +99,67 @@ func BindEnvAndSetDefault(key string, val interface{}, desc string) {
...
+	baseLogger, _ := loggerConfig.Build()
+	logger := baseLogger.Sugar()
+
+	return logger
+}
+
+func Listen(listen string, router http.Handler, logger *zap.SugaredLogger) {
+	if err := http.ListenAndServe(listen, router); err != nil {
+		logger.Errorw("HTTP listen error", "error", err)
+		os.Exit(1)
```
Comment (alexdias) [inline cmd/messenger/main.go]: I don't know if this is a good idea as it just kills all other goroutines without giving them a chance to cleanly stop

Comment (ThibaultLaurens) [reply cmd/messenger/main.go]: yeah, we will have proper error channel later

**Pull request merged**